plugins/woocommerce-blocks/tests/e2e/tests/checkout/checkout-block.shopper.block_theme.side_effects.spec.ts

```diff
@@ -474,3 +474,135 @@ test.describe( 'Checkout Form Errors', () => {
 		).toBeVisible();
 	} );
 } );
+
+test.describe( 'Billing Address Form', () => {
+	const blockSelectorInEditor = blockData.selectors.editor.block as string;
+
+	// To make sure the company field is visible in the billing address form, we need to enable it in the editor.
+	test.beforeEach( async ( { editor, admin, editorUtils } ) => {
+		await admin.visitSiteEditor( {
+			postId: 'woocommerce/woocommerce//page-checkout',
+			postType: 'wp_template',
+		} );
+		await editorUtils.enterEditMode();
+		await editor.openDocumentSettingsSidebar();
+		await editor.selectBlocks(
+			blockSelectorInEditor +
+				'  [data-type="woocommerce/checkout-shipping-address-block"]'
+		);
+
+		const checkbox = editor.page.getByRole( 'checkbox', {
+			name: 'Company',
+			exact: true,
+		} );
+		await checkbox.check();
+		await expect( checkbox ).toBeChecked();
+		await expect(
+			editor.canvas.locator(
+				'div.wc-block-components-address-form__company'
+			)
+		).toBeVisible();
+		await editorUtils.saveSiteEditorEntities();
+	} );
+
+	const shippingTestData = {
+		firstname: 'John',
+		lastname: 'Doe',
+		company: 'Automattic',
+		addressfirstline: '123 Easy Street',
+		addresssecondline: 'Testville',
+		country: 'United States (US)',
+		city: 'New York',
+		state: 'New York',
+		postcode: '90210',
+		phone: '01234567890',
+	};
+	const billingTestData = {
+		first_name: '',
+		last_name: '',
+		company: '',
+		address_1: '',
+		address_2: '',
+		country: 'United States (US)',
+		city: '',
+		state: 'New York',
+		postcode: '',
+		phone: '',
+	};
+
+	test( 'Ensure billing is empty and shipping address is filled for guest user', async ( {
+		frontendUtils,
+		page,
+		checkoutPageObject,
+	} ) => {
+		await frontendUtils.logout();
```

Comment:
👋 @tarunvijwani, just popping in to let you know that we shouldn't be logging the user out via the UI as it will stay logged out, affecting all the subsequent tests. Instead, let's use the dedicated guest user storage state file like in the test above:

https://github.com/woocommerce/woocommerce/blob/d5bcd7ccd7425eb3d125e0466043d95037a3d5b7/plugins/woocommerce-blocks/tests/e2e/tests/checkout/checkout-block.shopper.block_theme.side_effects.spec.ts#L415-L416

FYI, I am reverting it in https://github.com/woocommerce/woocommerce/pull/45624 as the tests are failing.

Reply:
@WunderBart Thank you for catching the bug and sharing the solution. TIL. 🙂